### PR TITLE
Adds missing endpoints

### PR DESCRIPTION
**Pull request by JakeBooher**
#### Why?
Required endpoints for our application are missing

#### How?
Adds an endpoint method for the following:

1. /contacts/search
2. /conversations/search

### CODE REVIEW

File: src/IntercomContacts.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    /**
+     * Returns list of Contacts that match search query.
+     *
+     * @see     https://developers.intercom.com/reference#search-for-contacts
+     * @param   array $options
+     * @return  stdClass
+     * @throws  Exception
+     */
+    public function searchContacts(array $options)
```
Comment (GabrielAnca) [inline src/IntercomContacts.php:18]: Considering `Contacts` is already the name of this class, I'd rename this to `search`

File: src/IntercomConversations.php [hunk 2]
```diff
@@ -35,6 +35,20 @@ public function getConversation($id, $options = [])
...
 
+    /**
+     * Returns list of Conversations that match search query.
+     *
+     * @see     https://developers.intercom.com/reference#search-for-conversations
+     * @param   array $options
+     * @return  stdClass
+     * @throws  Exception
+     */
+    public function searchConversations(array $options)
```
Comment (GabrielAnca) [inline src/IntercomConversations.php:46]: I would also rename this to `search`. I see the rest of the methods include the word conversations, but to be honest I would prefer to stick to `search` and keep the goal to rename the rest in a future version

**Comment by jourdan-qe** [comment]
Hi @GabrielAnca @JakeBooher,

Is it possible to add the "Retreive a contact by ID" route too please?

I need to get that quickly or I'll use directly the REST API with my own implementation.
So, could I contribute to your PR if you're running out of time?

**Comment by GabrielAnca** [comment]
Thank you so much for the PR @JakeBooher 🙇 I'll be closing this PR in favour of https://github.com/intercom/intercom-php/pull/306 - that one includes these changes + the change suggested by @jourdan-qe and has also some addtional contributions. Feel free to have a look at that PR and make your comments 🙌

**Comment by jourdan-qe** [comment]
@jourdan-qe Great! Thank you a lot

**Pull request closed without merging**